src/main/java/seedu/mtracker/filemanager/StockDecoder.java

```diff
@@ -4,13 +4,20 @@
 import seedu.mtracker.model.InstrumentManager;
 import seedu.mtracker.model.subinstrument.Stock;
 
+import java.time.LocalDate;
+
 public class StockDecoder extends InstrumentDecoder {
 
     public static final int STOCK_REMARKS_INDEX = 5;
+    protected static String decodedRemarks;
+
+    private static void decodeSpecificAttributes(String[] textSegment) {
+        decodedRemarks = textSegment[STOCK_REMARKS_INDEX];
+    }
 
     public static void addStockToList(String[] textSegment, InstrumentManager instrumentManager) {
         decodeGeneralAttributes(textSegment);
-        String decodedRemarks = textSegment[STOCK_REMARKS_INDEX];
+        decodeSpecificAttributes(textSegment);
         Instrument stock = new Stock(decodedName, decodedCurrPrice, decodedSentiment, decodedRemarks);
         setDoneStatus(decodedIsDone, stock);
         instrumentManager.addInstrument(stock);
```

Comment:
These lines can be abstracted out into InstrumentDecoder class as a new method like addDecodedInstrument(Instrument instrument).
This method should take in the new Instrument object (eg. new Stock(...) of line 21, new Etf(...), etc.)